thirdparty/curl/CMakeLists.txt

```diff
@@ -1,41 +1,112 @@
 cmake_minimum_required(VERSION 3.5.1)
```

Comment:
As an aside and not really relevant here, but after dealing with #1682 we can jump to 3.16. (Otoh I think feature-wise I'd want at least 3.12 but it'd be basically untested.)

Strictly speaking of course that can also be done separately if desired, but it's probably less effort overall that way.